packages/data-stores/src/plans/mock/mock-constants.ts

```diff
@@ -0,0 +1,477 @@
+};
+export const MOCK_FEATURE_RECURRING_PAYMENTS = {
+	id: 'recurring-payments',
+	name: 'Sell subscriptions (recurring payments)',
+	description: 'Accept one-time, monthly or annual payments on your website.',
+};
+export const MOCK_FEATURE_WORDADS = {
+	id: 'wordads',
+	name: 'WordAds',
+	description: 'Put your site to work and earn through ad revenue.',
+};
+
+// Feature groups (by type)
+export const MOCK_FEATURES_BY_TYPE_GENERAL: FeaturesByType = {
+	id: 'general',
+	name: null,
+	features: [
+		MOCK_FEATURE_CUSTOM_DOMAIN.id,
+		MOCK_FEATURE_LIVE_SUPPORT.id,
+		MOCK_FEATURE_PRIORITY_SUPPORT.id,
+	],
+};
+export const MOCK_FEATURES_BY_TYPE_COMMERCE: FeaturesByType = {
+	id: 'commerce',
+	name: 'Commerce',
+	features: [ MOCK_FEATURE_RECURRING_PAYMENTS.id ],
+};
+export const MOCK_FEATURES_BY_TYPE_MARKETING: FeaturesByType = {
+	id: 'marketing',
+	name: 'Marketing',
+	features: [ MOCK_FEATURE_WORDADS.id ],
+};
+
+// All plans details (from APIs)
+export const MOCK_PLAN_DETAILS_API: DetailsAPIResponse = {
+	groups: [
+		{
+			slug: 'personal',
+			name: 'Personal',
+		},
+	],
+	plans: [
+		{
+			support_priority: 1,
+			support_name: 'free',
+			groups: [ 'personal' ],
+			products: [
+				{
+					plan_id: 1,
+				},
+			],
+			name: 'WordPress.com Free',
+			short_name: 'Free',
+			nonlocalized_short_name: 'Free',
+			tagline: 'Mock free plan',
+			description:
+				'If you just want to start creating, get a free site and be on your way to publishing in less than five minutes.',
+			features: [ 'subdomain' ],
+			highlighted_features: [ 'Free plan highlighted feature' ],
+			storage: '3 GB',
+			icon: 'https://s0.wordpress.com/i/store/mobile/plans-free.png',
+		},
+		{
+			support_priority: 4,
+			support_name: 'premium',
+			groups: [ 'business' ],
+			products: [
+				{
+					plan_id: 1003,
+				},
+				{
+					plan_id: 1013,
+				},
+			],
+			name: 'WordPress.com Premium',
+			short_name: 'Premium',
+			nonlocalized_short_name: 'Premium',
+			tagline: 'Mock premium plan',
+			description:
+				'Build a unique website with advanced design tools, CSS editing, lots of space for audio and video, and the ability to monetize your site with ads.',
+			features: [ 'custom-domain', 'support-live', 'recurring-payments', 'wordads' ],
+			highlighted_features: [
+				MOCK_FEATURE_CUSTOM_DOMAIN.name,
+				MOCK_FEATURE_LIVE_SUPPORT.name,
+				'Premium plan highlighted feature',
+			],
+			storage: '13 GB',
+			icon: 'https://s0.wordpress.com/i/store/mobile/plans-premium.png',
+		},
+	],
+	features_by_type: [
+		MOCK_FEATURES_BY_TYPE_GENERAL,
+		MOCK_FEATURES_BY_TYPE_COMMERCE,
+		MOCK_FEATURES_BY_TYPE_MARKETING,
+	],
+	features: [
+		MOCK_FEATURE_CUSTOM_DOMAIN,
+		MOCK_FEATURE_LIVE_SUPPORT,
+		MOCK_FEATURE_PRIORITY_SUPPORT,
+		MOCK_FEATURE_RECURRING_PAYMENTS,
+		MOCK_FEATURE_WORDADS,
+	],
+};
+
+// Individual plan (from APIs)
+// @TODO: path_slug doesn't exist on monthly and free plans
```

Comment:
While gathering mock data from the API endpoints, I realised that the data for the free and the monthly-billed plans doesn't contain the `path_slug` property.

This is causing a TypeScript error, since we expect this value to be always defined (see `PricedAPIPlan` and, consequently, `PlanProduct` types), and has potential consequence in Gutenboarding, where we expect `path_slug` to always be defined.

I see two potential ways of getting around this:
1. The cleanest way would be to update the REST APIs and include `path_slug` also in the Free and monthly-billed plans
2. As an alternative, we could update the `PricedAPIPlan` type and mark `path_slug` as optional. Then, in the data-store, when building `PlanProduct` objects, we can use the "agnostic" slug so that we can ensure that the `PlanProduct` type always has a defined `path_slug`

Reply:
We're already doing the alternative (2) for _Free_ plan: https://github.com/Automattic/wp-calypso/blob/trunk/packages/data-stores/src/plans/resolvers.ts#L154

I don't have a strong opinion on this but whatever solution you implement, you should also make sure the bugs fixed in these PRs are not coming back 😅 :
- https://github.com/Automattic/wp-calypso/pull/49392
- https://github.com/Automattic/wp-calypso/pull/49426
As a bonus point, increasing unit test coverage to prevent future bugs like this would be awesome.

Reply:
Follow-up question: Do both plans need the `path_slug` prop? If not, then it might be a good idea to split up `PricedAPIPlan`, like we suggest doing for `DomainSuggestions` here #49540. 

I would definitely go for option 1 since we won't deviate from the API.

Reply:
> We're already doing the alternative (2) for Free plan: https://github.com/Automattic/wp-calypso/blob/trunk/packages/data-stores/src/plans/resolvers.ts#L154

We _sort of_ are, although TypeScript still thinks that a Free plan returned from the APIs should have a `path_slug` property

> I don't have a strong opinion on this but whatever solution you implement, you should also make sure the bugs fixed in these PRs are not coming back

I'll explore the two alternatives and make a more informed decicion

> And as a bonus point, increasing unit test coverage to prevent future bugs like these would be awesome.

Absolutely. I'll look into adding more tests to this PR too

Reply:
> Do both plans need the path_slug prop?

On a functional level, they kind of do. We have wordpress.com/new/free flow and wordpress.com/new/business flow and so on.